commitizen/changelog.py

```diff
@@ -72,6 +72,8 @@ def generate_tree_from_commits(
     commit_parser: str,
     changelog_pattern: str = defaults.bump_pattern,
     unreleased_version: Optional[str] = None,
+    change_type_map: Optional[Dict[str, str]] = None,
+    message_hook: Optional[Callable] = None,
```

Comment:
I'm not sure whether we should name it as `message_hook`. It looks like `changelog_hook` and makes me think it as a webhook intuitivly but it actually has a different functionality.

Reply:
mmmm :thinking: some alternatives I can think of:

- `per_commit_hook`
- `commit_hook`
- `changelog_line_hook`

any other ideas?

Reply:
Hmmm... My first though was `message_builder`. Or maybe `commit_preprocess_hook`?

Reply:
`commit_preprocess_hook`: gives me the impression of a precommit hook, or something unrelated to changelog. 

What about `changelog_message_builder_hook`? it may be long but goes to the point

Reply:
`changelog_message_builder_hook`  looks good to me.